src/rust/ide/enso-protocol/src/project_manager.rs

```diff
@@ -0,0 +1,382 @@
+//! Client library for the JSON-RPC-based Project Manager service.
+
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+use crate::prelude::*;
+
+use crate::common::UTCDateTime;
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_rpc_methods;
+use json_rpc::make_param_map;
+use json_rpc::make_arg;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// Event emitted by the Project Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ====================
+// === Notification ===
+// ====================
+
+/// Notification generated by the Project Manager.
```

Comment:
This doc is missing information what this notification is for and how it will be used. Especially I do not understand why the implementation is an empty enum

Reply:
But this explains, that this is notification by Project Manager. What it is should be covered in Engine documentation.

Reply:
I dont agree. What it is should be covered here as well. Even few sentences of explanation would clarify the picture a lot.

Reply:
Much better!